source/common/upstream/health_checker_impl.h

```diff
@@ -140,7 +141,7 @@ class HttpHealthCheckerImpl : public HealthCheckerImplBase {
 
   const std::string path_;
   const std::string host_value_;
-  absl::optional<std::string> service_name_;
+  Envoy::Matchers::StringMatcherPtr service_name_matcher_;
```

Comment:
Based on https://abseil.io/tips/123 I'm trying to see if `service_name_matcher_ ` should be a `absl::optional<Matchers::StringMatcherImpl>` or a `Envoy::Matchers::StringMatcherPtr`. Which one should we use?